### PR TITLE
[IOTDB-1673] CLI refactor and upgrade to JLine3

### PR DESCRIPTION
**Pull request by cornmonster**
1. Upgrade to Jline3.
2. Remove the CLI implementation for Windows since Jline3 is platform-independent.
3. Support persisted command history.
4. Support multi-line edition.
image

5. Exit by pressing CTRL+D or CTRL+C twice.
image

6. Add auto pair and autosuggestion widgets.
image

7. Add keyword highlighter and completer.
image

### CODE REVIEW

File: cli/pom.xml [hunk 1]
```diff
@@ -66,9 +66,18 @@
...
         </dependency>
         <dependency>
-            <groupId>jline</groupId>
+            <groupId>org.jline</groupId>
             <artifactId>jline</artifactId>
         </dependency>
+        <dependency>
+            <groupId>net.java.dev.jna</groupId>
+            <artifactId>jna</artifactId>
+        </dependency>
```
Comment (HTHou) [inline cli/pom.xml:75]: Hi, what is this dependency use for?

Comment (cornmonster) [reply cli/pom.xml:75]: To access the JVM's main terminal under the Windows system for Jline3 to provide advanced features.

Please see:
image

**Review by SteveYurongSu — APPROVED** [review]
LGTM :D

**Comment by HTHou** [comment]
Hi, would you like to update the version in LICENSE-binary?
https://github.com/apache/iotdb/blob/master/LICENSE-binary

**Comment by cornmonster** [comment]
> Hi, would you like to update the version in LICENSE-binary?
> https://github.com/apache/iotdb/blob/master/LICENSE-binary

No problem.

**Review by HTHou — APPROVED** [review]

**Pull request merged**